src/openqaoa-qiskit/backends/qaoa_qiskit_qpu.py

```diff
@@ -128,24 +131,38 @@ def qaoa_circuit(self, params: QAOAVariationalBaseParams) -> QuantumCircuit:
         qaoa_circuit: `QuantumCircuit`
             The final QAOA circuit after binding angles from variational parameters.
         """
+        parametric_circuit = deepcopy(self.parametric_circuit)
+
+        if self.append_state:
+            parametric_circuit = parametric_circuit.compose(self.append_state)
+
+        # only measure the problem qubits
+        if self.final_mapping is None:
+            parametric_circuit.measure(self.problem_reg, self.creg)
+        else:
+            for idx, qubit in enumerate(self.final_mapping[0 : len(self.problem_reg)]):
+                cbit = self.creg[idx]
+                parametric_circuit.measure(qubit, cbit)
+                
 
         angles_list = self.obtain_angles_for_pauli_list(self.abstract_circuit, params)
         memory_map = dict(zip(self.qiskit_parameter_list, angles_list))
-        circuit_with_angles = self.parametric_circuit.bind_parameters(memory_map)
+        new_parametric_circuit = parametric_circuit.bind_parameters(memory_map)
```

Comment:
also, the output of this function is not a parametric circuit, so why are calling it `new_parametric_circuit` as opposed to the older name?